README.md
# vera.cms

## Project setup
```
yarn install
```

### Compiles and hot-reloads for development
```
yarn serve
```

### Compiles and minifies for production
```
yarn build
```

### Lints and fixes files
```
yarn lint
```

### Customize configuration
See [Configuration Reference](https://cli.vuejs.org/config/).


CHANGELOG.md
# [1.6.0](https://github.com/averichev/vera.cms/compare/v1.5.0...v1.6.0) (2021-05-05)

### Features

- Raw html block for editor.js ([b2000f9](https://github.com/averichev/vera.cms/commit/b2000f90f71983e776442db68254af5cf6f0542b))

### Other

- Merge pull request #10 from averichev/dev-9-raw-html-block ([ebfd08b](https://github.com/averichev/vera.cms/commit/ebfd08bed73271aa743a5a891cbd51e2a51e350e)), closes [#10](https://github.com/averichev/vera.cms/issues/10)

# [1.5.0](https://github.com/averichev/vera.cms/compare/v1.4.0...v1.5.0) (2021-03-11)

### Features

- Description field for new page #7 ([8649013](https://github.com/averichev/vera.cms/commit/86490131c50b6821170e6246cd12231e4eacb3b9)), closes [#7](https://github.com/averichev/vera.cms/issues/7)

# [1.4.0](https://github.com/averichev/vera.cms/compare/v1.3.0...v1.4.0) (2021-03-11)

### Features

- Description field for page #7 ([36425dc](https://github.com/averichev/vera.cms/commit/36425dc40391779e0c0fac121d28caf5feca6eae)), closes [#7](https://github.com/averichev/vera.cms/issues/7)

### Improvements

- Darcula theme for code block ([f5d3af5](https://github.com/averichev/vera.cms/commit/f5d3af521c2c769482d84e9b3e5e6536f4ed5637))

### Other

- Merge pull request #8 from averichev/dev-7-description-field-for-page ([ae18ba5](https://github.com/averichev/vera.cms/commit/ae18ba5eae34a6021f8f3d7561b8135c0d95e491)), closes [#8](https://github.com/averichev/vera.cms/issues/8)

# [1.3.0](https://github.com/averichev/vera.cms/compare/v1.2.0...v1.3.0) (2021-03-11)

### Features

- Code block #5 ([690d9d7](https://github.com/averichev/vera.cms/commit/690d9d7a94471d0103ad48c9749d009b29f6f256)), closes [#5](https://github.com/averichev/vera.cms/issues/5)

### Other

- Merge pull request #6 from averichev/dev-5-code-block ([fe4a19e](https://github.com/averichev/vera.cms/commit/fe4a19e29459af71771736e599d497cd491170d5)), closes [#6](https://github.com/averichev/vera.cms/issues/6) [#5](https://github.com/averichev/vera.cms/issues/5)

# [1.2.0](https://github.com/averichev/vera.cms/compare/v1.1.0...v1.2.0) (2021-01-28)

### Chores

- **Prepared for deploy:** Merge ([4015a1f](https://github.com/averichev/vera.cms/commit/4015a1ff8b43fd195b15d74cc8d9200323b5f749))

### Features

- Prepared for deploy ([768cae4](https://github.com/averichev/vera.cms/commit/768cae43c5d7db62a7988b1114ca7f0f54656f58))

# [1.1.0](https://github.com/averichev/vera.cms/compare/v1.0.1...v1.1.0) (2021-01-26)

### Chores

- merge ([d1928c5](https://github.com/averichev/vera.cms/commit/d1928c50a860174259a0c9ed1ad8fe64a0eaaf0c))

### Features

- Authentication logic ([c522ec7](https://github.com/averichev/vera.cms/commit/c522ec7b64924996e0b5d62d3f23f4600acb011c))
- Pages CRUD ([31423bb](https://github.com/averichev/vera.cms/commit/31423bbb69adfec8a996a08a323abdd0d9a90727))
- Pages CRUD ([53282c7](https://github.com/averichev/vera.cms/commit/53282c7919d0c5351ba0e78fdd729f4b31a1cee4))

## [1.0.1](https://github.com/averichev/vera.cms/compare/v1.0.0...v1.0.1) (2021-01-23)

### Chores

- Automatic release ([1ae18cb](https://github.com/averichev/vera.cms/commit/1ae18cb1adc4987f8c1a55e755fbad9d1fb5cc7b))
- Automatic release ([130c389](https://github.com/averichev/vera.cms/commit/130c38924c1edde4eea7aea182749e6b406e630f))
- Automatic release ([7625a9d](https://github.com/averichev/vera.cms/commit/7625a9d7bc2eb4ae3f31c5d0d86fd2e2f89a0b1d))
- Automatic release ([226f55e](https://github.com/averichev/vera.cms/commit/226f55e5cd5b1cde8474619f92686dd2ff0888df))

### Other

- Merge branch 'master' of https://github.com/averichev/vera.cms ([91804b4](https://github.com/averichev/vera.cms/commit/91804b4849f7bb4a74b121e30f2a0255f12ec460))

# 1.0.0 (2021-01-22)

### Bug Fixes

- Editor.js and bootstrap ([c961c8e](https://github.com/averichev/vera.cms/commit/c961c8e4f1acb369f4291f6b5980db9951c8f705))

### Features

- Сохранение страницы ([f7d9033](https://github.com/averichev/vera.cms/commit/f7d9033731e78cba887b70cc042c83214245fa09))

### Other

- Create release.yml ([df8d577](https://github.com/averichev/vera.cms/commit/df8d577c829e20222ee19f6dc683f6b8335947f3))
- init commit ([f64cdd7](https://github.com/averichev/vera.cms/commit/f64cdd7a4fef3bf107a64f65a9339c7b30eed5c8))
- init ([3a805ac](https://github.com/averichev/vera.cms/commit/3a805ac2e95ce5f4b6aaca1e5ed7959639e4d1f0))


eslintrc.js
module.exports = {
  root: true,
  env: {
    node: true,
    jquery: true
  },
  extends: [
    'plugin:vue/essential',
    'plugin:vue/recommended',
    'eslint:recommended'
  ],
  rules: {
    'comma-dangle': ['warn', 'only-multiline'],
    complexity: ['warn', 5],
    'max-depth': ['warn', 3],
    'spaced-comment': ['error', 'always'],
    'max-lines': ['warn', 200],
    'max-lines-per-function': ['warn', 20],
    'max-nested-callbacks': ['warn', 2],
    'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-useless-escape': 'off',
    'vue/html-indent': ['warn', 4, {
      baseIndent: 1
    }],
    'vue/script-indent': ['warn', 2, {
      baseIndent: 1,
      switchCase: 1
    }]
  },
  overrides: [{
    files: ['*.vue'],
    rules: {
      indent: 'off',
      'max-lines': ['warn', 400]
    }
  }, {
    files: ['vue.config.js'],
    rules: {
      'max-lines-per-function': 'off'
    }
  }],
  parserOptions: {
    parser: 'babel-eslint'
  }
}


src/plugins/httpClient.js
import axios from 'axios'

function createAxiosInstance(){
  const baseUrl = process.env.VUE_APP_BASE_URL;
  let headers = {
    Accept: 'application/json',
    'Content-Type': 'application/json',
  };
  const token = localStorage.getItem('vera.token');
  if (token) {
    headers.Authorization = 'Bearer '+ token
  }
  return axios.create({
    baseURL: baseUrl,
    headers: headers,
  })
}

function install(pVue) {
  if (install.installed) return;
  install.installed = true;
  const client = createAxiosInstance();
  pVue.prototype.$httpClient = {
    get(path){
      return client.get(path)
    },
    post(path, data){
      return client.post(path, data)
    }
  }
}

let plugin = {
  install: install
};
let GlobalVue = null;
if (typeof window !== 'undefined') {
  GlobalVue = window.Vue;
} else if (typeof global !== 'undefined') {
  GlobalVue = global.Vue;
}
if (GlobalVue) {
  GlobalVue.use(plugin);
}

export {install};
export default plugin;


.eslintrc.js
module.exports = {
    root: true,
    env: {
        node: true,
        jquery: true,
    },
    extends: [
        'plugin:vue/essential'
    ],
    rules: {
        'comma-dangle': ['warn', 'only-multiline'],
        complexity: ['warn', 5],
        'max-depth': ['warn', 3],
        'spaced-comment': ['error', 'always'],
        'max-lines': ['warn', 200],
        'max-lines-per-function': ['warn', 20],
        'max-nested-callbacks': ['warn', 2],
        'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
        'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
        'no-useless-escape': 'off',
        'vue/html-indent': ['warn', 4, {
            baseIndent: 1,
        }],
        'vue/script-indent': ['warn', 2, {
            baseIndent: 1,
            switchCase: 1,
        }],
    },
    overrides: [{
        files: ['*.vue'],
        rules: {
            indent: 'off',
            'max-lines': ['warn', 400],
        },
    }, {
        files: ['vue.config.js'],
        rules: {
            'max-lines-per-function': 'off',
        },
    }],
    parserOptions: {
        parser: 'babel-eslint',
    },
}
